privacyidea/lib/smsprovider/FirebaseProvider.py

```diff
@@ -155,8 +157,13 @@ def submit_message(self, firebase_token, data):
                        }
             }
 
+        proxies = {}
+        if self.smsgateway.option_dict.get(FIREBASE_CONFIG.HTTP_PROXY):
+            proxies["http"] = self.smsgateway.option_dict.get(FIREBASE_CONFIG.HTTP_PROXY)
+        if self.smsgateway.option_dict.get(FIREBASE_CONFIG.HTTPS_PROXY):
+            proxies["https"] = self.smsgateway.option_dict.get(FIREBASE_CONFIG.HTTPS_PROXY)
         url = FIREBASE_URL_SEND.format(self.smsgateway.option_dict.get(FIREBASE_CONFIG.PROJECT_ID))
-        resp = requests.post(url, data=json.dumps(fcm_message), headers=headers)
+        resp = requests.post(url, data=json.dumps(fcm_message), headers=headers, proxies=proxies)
```

Comment:
As the ``url`` is always created from the fixed ``FIREBASE_URL_SEND``, which is always *https*, we might as well drop the settings for `` proxies["http"]``, since these are not necessary.